src/System.Private.CoreLib/shared/System/Exception.cs

```diff
@@ -140,27 +140,22 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
         }
 
         public override string ToString()
-        {
-            return ToString(true, true);
-        }
-
-        private string ToString(bool needFileLineInfo, bool needMessage)
         {
             string s = GetClassName();
 
-            string? message = (needMessage ? Message : null);
+            string? message = Message;
             if (!string.IsNullOrEmpty(message))
             {
                 s += ": " + message;
             }
 
             if (_innerException != null)
             {
-                s = s + " ---> " + _innerException.ToString(needFileLineInfo, needMessage) + Environment.NewLine +
+                s = s + " ---> " + _innerException.ToString() + Environment.NewLine +
                 "   " + SR.Exception_EndOfInnerExceptionStack;
             }
 
-            string? stackTrace = GetStackTrace(needFileLineInfo);
+            string? stackTrace = GetStackTrace(needFileInfo: true);
```

Comment:
The non-virtual internal `GetStackTrace` method has the same problem. It does the same thing as the virtual public `StackTrace` property, except that it cannot be over-ridden. Since we are always calling it with `needFileInfo: true` now, we do not need it anymore and we can just call the virtual `StackTrace` property.

(The non-virtual internal `GetStackTrace` method was introduced at the same time as the non-virtual internal `ToString` with extra argument. This change rolled back one half of the refactoring, but not the other half.)